doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
If you take a look at the existing documentation, when creating a new style, we do a POST to `/styles&name=foo` (for example). From what I can determine, this is done to maintain RESTful behaviour, as it may be the case that the style foo already extists. In order to be consistent with the existing behavior, we should stick to this convention, as demonstrated in my example above.